traffic_ops/traffic_ops_golang/server/update.go

```diff
@@ -112,42 +112,161 @@ func UpdateHandler(w http.ResponseWriter, r *http.Request) {
+
+	}
+
+	// Validate and parse parameters before attempting to apply them (don't want to partially apply various status before an error)
+	// Timestamps
+	if hasConfigUpdateTimeParam {
+		configUpdateTime, err := time.Parse(time.RFC3339Nano, configUpdateTimeParam)
+		if err != nil {
+			return nil, errors.New("query parameter 'config_update_time' must be valid RFC3339Nano format")
+		}
+		paramValues.configUpdateTime = &configUpdateTime
+	}
+
+	if hasRevalidateUpdateTimeParam {
+		revalUpdateTime, err := time.Parse(time.RFC3339Nano, revalidateUpdateTimeParam)
+		if err != nil {
+			return nil, errors.New("query parameter 'revalidate_update_time' must be valid RFC3339Nano format")
+		}
+		paramValues.revalUpdateTime = &revalUpdateTime
+	}
+
+	if hasConfigApplyTimeParam {
+		configApplyTime, err := time.Parse(time.RFC3339Nano, configApplyTimeParam)
+		if err != nil {
+			return nil, errors.New("query parameter 'config_apply_time' must be valid RFC3339Nano format")
+		}
+		paramValues.configApplyTime = &configApplyTime
+	}
+
+	if hasRevalidateApplyTimeParam {
+		revalApplyTime, err := time.Parse(time.RFC3339Nano, revalidateApplyTimeParam)
+		if err != nil {
+			return nil, errors.New("query parameter 'revalidate_apply_time' must be valid RFC3339Nano format")
+		}
+		paramValues.revalApplyTime = &revalApplyTime
+	}
+
+	// Booleans
+	configUpdatedBool := strings.ToLower(configUpdatedBoolParam)
+	revalUpdatedBool := strings.ToLower(revalUpdatedBoolParam)
+
+	if hasConfigUpdatedBoolParam && configUpdatedBool != `t` && configUpdatedBool != `true` && configUpdatedBool != `f` && configUpdatedBool != `false` {
+		return nil, errors.New("query parameter 'updated' must be 'true' or 'false'")
+	}
+	if hasRevalUpdatedBoolParam && revalUpdatedBool != `t` && revalUpdatedBool != `true` && revalUpdatedBool != `f` && revalUpdatedBool != `false` {
+		return nil, errors.New("query parameter 'reval_updated' must be 'true' or 'false'")
+	}
+
+	strToBool := func(s string) bool {
+		return strings.HasPrefix(s, "t")
+	}
+
+	if hasConfigUpdatedBoolParam {
+		updateBool := strToBool(configUpdatedBool)
+		paramValues.configUpdateBool = &updateBool
+	}
+
+	if hasRevalUpdatedBoolParam {
+		revalUpdatedBool := strToBool(revalUpdatedBool)
+		paramValues.revalUpdateBool = &revalUpdatedBool
+	}
+	return &paramValues, nil
+}
+
+// setUpdateStatuses set timestamps for config update/apply and revalidation
+// update/apply. If any value is nil, no changes occur
+func setUpdateStatuses(tx *sql.Tx, serverID int64, values updateValues) error {
+
+	if values.configUpdateTime != nil {
+		if err := dbhelpers.QueueUpdateForServerWithTime(tx, serverID, *values.configUpdateTime); err != nil {
+			return fmt.Errorf("setting config apply time: %w", err)
+		}
+	}
+
+	if values.revalUpdateTime != nil {
+		if err := dbhelpers.QueueRevalForServerWithTime(tx, serverID, *values.revalUpdateTime); err != nil {
+			return fmt.Errorf("setting reval apply time: %w", err)
+		}
+	}
```

Comment:
I thought the server (TO) was supposed to ignore these times? `t3c` wouldn't be sending the update_times anyways, right?